apps/crn-server/test/data-providers/contentful/research-output.data-provider.test.ts

```diff
@@ -0,0 +1,1541 @@
+    });
+
+    test('Should default asapFunded to undefined when missing', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.asapFunded = null;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.asapFunded).not.toBeDefined();
+    });
+
+    test('Should default asapFunded "Not Sure" option to undefined', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.asapFunded = 'Not Sure';
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.asapFunded).not.toBeDefined();
+    });
+
+    test('Should default usedInPublication to undefined when missing', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.usedInAPublication = null;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.usedInPublication).not.toBeDefined();
+    });
+
+    test('Should default usedInPublication "Not Sure" option to undefined', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.usedInAPublication = 'Not Sure';
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.usedInPublication).not.toBeDefined();
+    });
+
+    test('Should default authors to an empty array when missing', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.authorsCollection = null;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.authors).toEqual([]);
+    });
+
+    test('Should default related research to an empty array when missing', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
+      researchOutputs.relatedResearchCollection = null;
+      researchOutputs.linkedFrom!.researchOutputsCollection = null;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        researchOutputs,
+      });
+
+      const result = await researchOutputDataProvider.fetchById('1');
+
+      expect(result!.relatedResearch).toEqual([]);
+    });
+
+    test('Should combine inbound and outbound related research links', async () => {
+      const researchOutputs = getContentfulResearchOutputGraphqlResponse();
```

Comment:
right now you're depending on the fixture for this test, i like to be explicit when testing different edge cases, even if the fixture data already works for us